### PR TITLE
[Runtime] Fixes the issue with missing `CFP_t`

### PR DESCRIPTION
**Pull request by maliasadi**
This fixes the issue with building `StateVectorLQubitDynamic` against the master branch of the lightning monorepo after merging PR https://github.com/PennyLaneAI/pennylane-lightning/pull/499.

### CODE REVIEW

**Comment by dime10** [comment]
Thanks @maliasadi, let's add a changelog and add the reference commit hash (from the lightning repo) that this fix is targeting to our makefile (instead of the current "master" tag). This will allow us to go back through our git history in the future and still have reproduceable runtime builds :)

File: doc/changelog.md [hunk 1]
```diff
@@ -12,6 +12,11 @@
 
 <h3>Bug fixes</h3>
 
+* Fixes the issue with missing `CFP_t` in `StateVectorLQubitDynamic` when building against the master
+  branch of PennyLane-Lightning. This issue introduced in [PR 499](https://github.com/PennyLaneAI/pennylane-lightning/pull/499).
+
```
Comment (maliasadi) [inline doc/changelog.md:17]:
```suggestion
```

File: doc/changelog.md [hunk 2]
```diff
@@ -12,6 +12,11 @@
 
 <h3>Bug fixes</h3>
 
+* Fixes the issue with missing `CFP_t` in `StateVectorLQubitDynamic` when building against the master
+  branch of PennyLane-Lightning. This issue introduced in [PR 499](https://github.com/PennyLaneAI/pennylane-lightning/pull/499).
+  
+  [(#322)](https://github.com/PennyLaneAI/catalyst/pull/322)
```
Comment (dime10) [inline doc/changelog.md:18]:
```suggestion
  [(#322)](https://github.com/PennyLaneAI/catalyst/pull/322)
```

**Review by dime10 — APPROVED** [review]
Great, thanks for the quick fix 💪

**Pull request merged**